firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -80,42 +84,40 @@ internal constructor(
     )
   }
 
-  private fun initiateSessionStart() {
+  private suspend fun initiateSessionStart() {
     val sessionDetails = sessionGenerator.generateNewSession()
 
-    sessionStartScope.launch {
-      val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
-
-      if (subscribers.isEmpty()) {
-        Log.d(
-          TAG,
-          "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
-        )
-        return@launch
-      }
+    val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
 
-      if (subscribers.values.none { it.isDataCollectionEnabled }) {
-        Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
-        return@launch
-      }
+    if (subscribers.isEmpty()) {
+      Log.d(
+        TAG,
+        "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
+      )
+      return
+    }
 
-      Log.d(TAG, "Data Collection is enabled for at least one Subscriber")
+    if (subscribers.values.none { it.isDataCollectionEnabled }) {
+      Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
+      return
+    }
 
-      subscribers.values.forEach { subscriber ->
-        if (subscriber.isDataCollectionEnabled) {
-          subscriber.onSessionChanged(SessionSubscriber.SessionDetails(sessionDetails.sessionId))
-        }
-      }
+    Log.d(TAG, "Data Collection is enabled for at least one Subscriber")
 
-      if (!sessionGenerator.collectEvents) {
-        Log.d(TAG, "Sessions SDK has sampled this session")
-        return@launch
+    subscribers.values.forEach { subscriber ->
+      if (subscriber.isDataCollectionEnabled) {
```

Comment:
We should still notify subscribers when their data collection is turned off, and when all subscriber data collection is turned off. 

We can move this up to line 99 and remove this if statement.

It might also be good to put a comment here stating that we should notify subscribers regardless of sampling and data collection state

Reply:
Done